File: java/socketd/src/main/java/org/noear/socketd/transport/core/listener/RouteSelector.java
package org.noear.socketd.transport.core.listener;

/**
 * 路由选择器
 *
 * @author noear
 * @since 2.0
 */
public interface RouteSelector<T> {
    /**
     * 选择
     *
     * @param route 路由
     */
    T select(String route);

    /**
     * 放置
     *
     * @param route  路由
     * @param target 目标
     */
    void put(String route, T target);

    /**
     * 移除
     *
     * @param route 路由
     */
    void remove(String route);

    /**
     * 数量
     */
    int size();
}

File: python/socketd/socketd/transport/core/listener/RouteSelector.py
import abc

from socketd.transport.core import Listener

# 路由选择器
class RouteSelector:
    # 选择
    @abc.abstractmethod
    def select(self, route:str)->Listener:...

    # 放置
    @abc.abstractmethod
    def put(self, route:str, target:Listener):...

    # 移除
    @abc.abstractmethod
    def remove(self, route:str):...

    # 数量
    @abc.abstractmethod
    def size(self)->int:...